runtime/doc/lua-plugin.txt

```diff
@@ -0,0 +1,432 @@
+*lua-plugin.txt*                     Nvim
+
+                            NVIM REFERENCE MANUAL
+
+		  Guide to developing Lua plugins for Nvim
+
+
+                                       Type |gO| to see the table of contents.
+
+==============================================================================
+Introduction                                                       *lua-plugin*
+
+This is a guide for getting started with Nvim plugin development. It is not
+intended as a set of rules, but as a collection of recommendations for good
+practices.
+
+For a guide to using Lua in Nvim, please refer to |lua-guide|.
+
+==============================================================================
+Type safety                                            *lua-plugin-type-safety*
+
+Lua, as a dynamically typed language, is great for configuration. It provides
+virtually immediate feedback.
+But for larger projects, this can be a double-edged sword, leaving your plugin
+susceptible to unexpected bugs at the wrong time.
+
+You can leverage LuaCATS https://luals.github.io/wiki/annotations/
+annotations, along with lua-language-server https://luals.github.io/ to catch
+potential bugs in your CI before your plugin's users do.
+
+------------------------------------------------------------------------------
+Tools						 *lua-plugin-type-safety-tools*
+
+- lua-typecheck-action https://github.com/marketplace/actions/lua-typecheck-action
+- llscheck https://github.com/jeffzi/llscheck
```

Comment:
This seems absolutely broken. Have you tried this and gotten it to work? It gives completely different results if you run it for a directory `llscheck scripts` vs `llscheck scripts/*` and both of the are wrong, the `--configpath` parameter seems to do nothing. It's impressively bad. What is this doing here?

Reply:
This one hasn't been touched in a while, so yeah it's very possibly broken.

Reply:
Perhaps my weird results with llscheck was due to our .luarc.json being incorrectly configured. Can keep in mind for later.